core/corehttp/gateway_handler.go

```diff
@@ -383,6 +383,15 @@ func (i *gatewayHandler) serveFile(w http.ResponseWriter, req *http.Request, nam
 		}
 	}
 
+	buf := make([]byte, 512)
+	_, _ = content.Read(buf)
```

Comment:
This is the only way to do this but we should be calling `io.ReadFull`. Otherwise, we could read a single byte.

Reply:
Okay. as you say.

Maybe I'm just confused because, according to the docs, the main difference between [Read](https://golang.org/pkg/io/#Reader) and [ReadFull](https://golang.org/pkg/io/#ReadFull) seems to be that `Read` **may** read up to `len(buf)` while `ReadFull` **must** read exactly `len(buf)`.

Also, I have tested the following code\* with both `.mp4` and `.html` files and it seemed to work just fine. _Like, it was reading more than a single byte._

(\* I couldn't build/test the actual codebase for technical reasons. Sorry for the annoyance.)

```go
func main() {
	w, err := os.Open("clip.mp4")
	_ = err

	buf := make([]byte, 512)
	_, _ = w.Read(buf)
	contentType := http.DetectContentType(buf)
	fmt.Println("Type is", contentType)
	if strings.HasPrefix(contentType, "text/html;") {
		contentType = "text/html"
	}

	fmt.Println("Type is", contentType)
}
```

Reply:
> Maybe I'm just confused because, according to the docs, the main difference between Read and ReadFull seems to be that Read may read up to len(buf) while ReadFull must read exactly len(buf).

Exactly. The issue is that `Read` may _also_ not read the full buffer (e.g., it can return after reading a single byte).

> Like, it was reading more than a single byte.

It probably was, that's just not something we can rely on.